zbus/src/abstractions/process.rs

```diff
@@ -31,12 +47,117 @@ impl Command {
 
     /// Executes the command as a child process, waiting for it to finish and
     /// collecting all of its output.
+    #[cfg(target_os = "macos")]
     pub async fn output(&mut self) -> Result<Output, Error> {
         self.0.output().await
     }
+
+    /// Sets configuration for the child process's standard input (stdin) handle.
+    pub fn stdin<T: Into<Stdio>>(&mut self, cfg: T) -> &mut Command {
+        self.0.stdin(cfg);
+        self
+    }
+
+    /// Sets configuration for the child process's standard output (stdout) handle.
+    pub fn stdout<T: Into<Stdio>>(&mut self, cfg: T) -> &mut Command {
+        self.0.stdout(cfg);
+        self
+    }
+
+    /// Sets configuration for the child process's standard error (stderr) handle.
+    pub fn stderr<T: Into<Stdio>>(&mut self, cfg: T) -> &mut Command {
+        self.0.stderr(cfg);
+        self
+    }
+
+    /// Executes the command as a child process, returning a handle to it.
+    pub fn spawn(&mut self) -> Result<Child, Error> {
+        Ok(Child::from(self.0.spawn()?))
+    }
+}
+
+/// A wrapper around the child API of the underlying async runtime.
+pub struct Child(
```

Comment:
* Addition of this type should either be done as a separate commit or as part of `Command` type addition (with the commit message adjusted appropriately).
* Please make sure only parts that are needed by tests, are declared `pub`. Others should be `pub(crate)`. If we add a `pub` API, please mark it as `#[doc(hidden)]`.